broker-core/src/test/java/io/zeebe/broker/workflow/message/MessageCorrelationTest.java

```diff
@@ -518,6 +519,50 @@ public void testBoundaryMessageEventLifecycle() {
             tuple("msg1", WorkflowInstanceIntent.EVENT_TRIGGERED));
   }
 
+  @Test
+  public void shouldCorrelateAndTakeFlowForEachMessageWithoutContinuing() {
```

Comment:
I don't like the name because it's not clear that you're testing the non-interrupting boundary event.

```suggestion
  public void shouldCorrelateToNonInterruptingBoundaryEvent() {
```